pkg/controller/daemon/controller.go

```diff
@@ -671,7 +671,7 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 		return err
 	}
 	dsNeedsSync := dsc.expectations.SatisfiedExpectations(dsKey)
-	if dsNeedsSync {
+	if dsNeedsSync && ds.DeletionTimestamp == nil {
```

Comment:
For RC,  I also need to check DeletionTimestamp!=nil before adopting orphaned pods. daemon controller may need to do the same.